include/AST/Types/ArrayTy.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef GLU_AST_TYPES_ARRAYTY_HPP_
+#define GLU_AST_TYPES_ARRAYTY_HPP_
+
+#include "Types/TypeBase.hpp"
+#include <optional>
+
+namespace glu::types {
+
+/// @brief Represents an array type.
+class ArrayTy : public TypeBase {
+    TypeKind const _dataKind;
+    enum ArrayMode { Dynamic, Static } _mode;
+    std::optional<std::size_t> _size;
+
+public:
+    /// @brief Constructor for the ArrayTy class.
+    /// @param dataKind The kind of the data that the array will hold.
+    /// @param mode The mode of the array (dynamic or static).
+    /// @param size The size of the array if the mode is static.
+    ArrayTy(
+        TypeKind dataKind, ArrayMode mode = ArrayMode::Dynamic,
```

Comment:
why dynamic by default ?

Reply:
I think that the fact that only static mode requires size makes it impossible to be static by default. And I think that having dynamic as a default mode makes sense because it doesn't require any arguments besides the dataKind.